l10n_th_partner/models/res_partner.py

```diff
@@ -15,6 +15,15 @@ class ResPartner(models.Model):
         string="Company name", inverse="_inverse_company_name", index=True
     )
 
+    @api.model
+    def create(self, vals):
+        """Add inverted company names at creation if unavailable."""
+        context = dict(self.env.context)
+        name = vals.get("name", context.get("default_name"))
+        if vals.get("is_company", False) and name:
+            vals["company_name"] = name
+        return super(ResPartner, self.with_context(context)).create(vals)
```

Comment:
Why do you send context in super ?

Reply:
Thank you, I misunderstand. I changed this code.